erpnext/accounts/doctype/pos_invoice/test_pos_invoice.py

```diff
@@ -377,6 +387,15 @@ def test_without_payment(self):
 		inv.payments = []
 		self.assertRaises(frappe.ValidationError, inv.insert)
 
+	def test_partial_payment(self):
+		pos_inv = create_pos_invoice(rate=10000, do_not_save=1)
+		pos_inv.append(
+			"payments",
+			{"mode_of_payment": "Cash", "account": "Cash - _TC", "amount": 9000},
+		)
+		pos_inv.insert()
+		self.assertRaises(frappe.ValidationError, pos_inv.submit)
```

Comment:
While testing a specific validation define an error class for the validation to ensure that the exact validation is raised.